caktus/philly-hip
# Upgrade NodeJS from version 12 to version 16

This PR upgrades NodeJS from version 12 to version 16. Changes include:
- updating and adding new dependencies and scripts in `package.json`
- removing `webpack_loader` dependency
- adding a new `gulpfile.js` file
- adding a new `webpack.config.js` file (and deleting 3 files- `webpack.local.config.js`, `webpack.prod.config.js`, and `webpack.base.config.js`)
- adding a `webpack_entry.js` file
- updating `Dockerfile` to use Node 16
- updating README to reflect NodeJS version change

To test locally:
- switch to using node 16
`nvm use 16`
- install new JS libraries
`npm install`
- run `make run-dev` and check
  - does editing JS files auto-generate a new bundle and show up locally?
  - does editing SCSS files auto-generate a new bundle and show up locally?
- run `npm run build` and check
  - does running the server (`python manage.py runserver`) have the correct JS and SCSS?
  - does editing JS files and running `npm run build` generate new bundle?
  - does editing SCSS files and running `npm run build` generate new bundle?
- click through site locally to determine it works like before (specifically in regards to JavaScript)
